ckanext/ogdchcommands/admin_logic.py

```diff
@@ -0,0 +1,127 @@
+from ckan.plugins.toolkit import side_effect_free, get_or_bust
+import ckan.model as model
+import ckan.plugins.toolkit as tk
+
+log = logging.getLogger(__name__)
+
+
+@side_effect_free
+def ogdch_reindex(context, data_dict):
+    current_user = context.get('user')
+    if not authz.is_sysadmin(current_user):
+        return "not authorized"
+    package_id = data_dict.get('id')
+    only_missing = data_dict.get('only_missing')
+
+    try:
+        rebuild_search_index(package_id=package_id, only_missing=only_missing)
+    except Exception as e:
+        log.error(str(e))
+        log.error(traceback.format_exc())
+        return {
+            'msg': "an error occured",
+            'error': str(e),
+            'traceback': traceback.format_exc()
+        }
+    return "Success: search index was rebuild"
+
+
+@side_effect_free
+def ogdch_check_indexing(context, data_dict):
+    current_user = context.get('user')
+    if not authz.is_sysadmin(current_user):
+        return "not authorized"
+
+    try:
+        package_query = query_for(model.Package)
+
+        log.debug("Checking packages search index...")
+        pkgs_q = model.Session.query(model.Package).filter_by(
+            state=model.State.ACTIVE)
+        pkgs = set([pkg.id for pkg in pkgs_q])
+        indexed_pkgs = \
+            set(package_query.get_all_entity_ids(max_results=len(pkgs)))
+        pkgs_not_indexed = pkgs - indexed_pkgs
+        return ("there are {} packages not indexed"
+                .format(len(pkgs_not_indexed)))
+    except Exception:
+        return "An Error occured"
+
+
+@side_effect_free
+def ogdch_check_field(context, data_dict):
+    current_user = context.get('user')
+    if not authz.is_sysadmin(current_user):
+        return "not authorized"
+    field = get_or_bust(data_dict, 'field')
+    if not field:
+        return "please provide a field name with field="
+
+    results = []
+    for package in _search_for_datasets(context):
+        field_data_raw = package.get(field)
+        field_data = ''
+        if field_data_raw:
+            try:
+                field_data = json.loads(field_data_raw)
+            except:
+                pass
+        result = {
+            'name': package.get('name'),
+            'field': field,
+            'field_stored': field_data_raw,
+            'field_data': field_data,
+        }
+        results.append(result)
+
+    return {
+        'msg': "So many datasets have been found:",
+        'count': len(results),
+        'pkgs': results,
+    }
+
+
+def _search_for_datasets(context):
+    rows = 500
+    page = 0
+    result_count = 0
+    fq = "dataset_type:(dataset)"
+    processed_count = 0
+    log.error("search for datasets")
+    while page == 0 or processed_count < result_count:
+        try:
+            log.error("now searching")
+            page = page + 1
+            start = (page - 1) * rows
+            log.error(fq)
+            log.error(rows)
+            log.error(start)
```

Comment:
These error logs seem unnecessary - can we get rid of them, please? There are some others in the other functions in this file as well.